TransferMethod/Sources/View/Widget/TextWidget.swift

```diff
@@ -151,66 +144,58 @@ class TextWidget: AbstractWidget {
         return inputText
     }
 
-    // swiftlint:disable function_parameter_count
-    private func applyFormatForPatternCharacter(currentPatternCharacter: [Character],
-                                                currentText: String,
-                                                currentTextCharacter: String,
-                                                currentTextIndex: inout String.Index,
+    private func applyFormatForPatternCharacter(currentText: String,
                                                 finalText: inout String,
-                                                isEscapedCharacter: inout Bool,
+                                                index: inout CurrentIndex,
                                                 pattern: String,
-                                                patternIndex: inout String.Index,
                                                 patternCharactersToBeWritten: inout String) {
-        switch currentPatternCharacter.first {
+        let currentPatternCharacter = getCharacterAt(index.patternIndex, pattern)
```

Comment:
just use `let currentPatternCharacter = pattern[index.patternIndex]` instead of

Reply:
not supported by swift